### ISSUE TITLE
Powers of ten quartiles has very low accuracy

### ISSUE THREAD

**Issue by mccalluc** [opendp/opendp#2644]
Motivated by the Dataverse usecase: They have incoming CSVs with limited data dictionaries, potentially containing PII, potentially with hundreds of columns, and they would be interested in a tool that would tell them a little about the shape of the data, without looking directly at the data. I had hoped that extreme percentiles, with candidates taken from powers of ten, would work. Last conversation suggested that quartiles would be better.

I have a proposed analysis now, but even applying it to the fake class dataset, the results seem bad. Am I making a mistake here, or is this approach just not going to work?

<details>
<summary>1000 lines, representing 100 students with 10 grades each</summary>

```
$ head -n 1 /Users/chuckmccallum/.dp-wizard/demo.csv | head
student_id,class_year_str,hw_number,grade,self_assessment
$ sort --random-sort /Users/chuckmccallum/.dp-wizard/demo.csv
| head
59,sophomore,6,84,0
84,sophomore,3,91,1
56,sophomore,5,86,0
100,sophomore,2,79,0
100,sophomore,8,83,0
30,senior,10,84,0
18,first year,3,95,1
97,senior,5,91,1
12,sophomore,2,94,0
27,sophomore,9,99,0
```

</details>

<details>
<summary>Power of ten quartiles on numeric columns</summary>

```python
import re
import opendp.prelude as dp
import polars as pl
from pprint import pprint

dp.enable_features("contrib")


def get_stem(name):
    return re.sub(r"\d*$", "", name)

for i in range(5):
    data=pl.scan_csv(
        "/Users/chuckmccallum/.dp-wizard/demo.csv",
    ).with_columns()

    stems = set()
    str_columns = []
    num_columns = []
    for k, v in data.collect_schema().items():
        stem = get_stem(k)
        # For a high-level summary, we'll assume all the columns with the same stem are similar.
        # If we make fewer queries, we can return more accurate results.
        if stem not in stems:
            stems.add(stem)
            target = num_columns if v.is_numeric() else str_columns
            target.append(k)

    privacy_unit = dp.unit_of(contributions=10)
    privacy_loss = dp.loss_of(epsilon=1.0)

    summary_context = dp.Context.compositor(
        data=data,
        privacy_unit=privacy_unit,
        privacy_loss=privacy_loss,
        split_evenly_over=len(str_columns) + 2 * len(num_columns),
        margins=[
            dp.polars.Margin(max_length=100) # TODO: Why is this required?
        ],
    )

    powers_of_ten = [0] + [10**n for n in range(7)]
    num_results = {}

    for col in num_columns:
        num_results[col] = [
            summary_context.query().select(
                pl.col(col).dp.quantile(alpha, powers_of_ten)
            ).release().collect().item()
            for alpha in [0.25, 0.75]
        ]

    pprint(num_results, width=40)    
```

</details>

<details><summary>Results don't look good</summary>

```
{'grade': [100, 10000],
 'hw_number': [10, 10000],
 'self_assessment': [0, 1000],
 'student_id': [1, 10000]}
{'grade': [1000000, 10],
 'hw_number': [1, 10],
 'self_assessment': [10000, 1000],
 'student_id': [100000, 1000000]}
{'grade': [1, 100000],
 'hw_number': [1, 100],
 'self_assessment': [10000, 1000000],
 'student_id': [100000, 1000]}
{'grade': [0, 100000],
 'hw_number': [1000, 1000000],
 'self_assessment': [1, 100000],
 'student_id': [100, 100]}
{'grade': [10000, 100],
 'hw_number': [1, 100],
 'self_assessment': [10, 100],
 'student_id': [100000, 100]}
```

</details>

Real data sets may have more rows, but from what I've looked at, they also have many more columns: The aspect ratio only gets worse.

If there isn't something obviously wrong, I'd like to get back to them soon, just so they aren't hoping we'll come through with something.
